src/routes/player/PlayQuiz.tsx

```diff
@@ -0,0 +1,92 @@
+import { Link } from "react-router-dom";
+import DisplayQuestion from "../../components/player/DisplayQuestion";
+import { useEffect } from "react";
+import { socket } from "../../socket";
+import { WS_EVENTS } from "../../socket/events";
+import { useCountdown } from "usehooks-ts";
+import { state$ } from "../../state";
+import ShowAnswer from "../../components/player/ShowAnswer";
+import WaitingForOtherPlayers from "../../components/player/WaitingForOtherPlayers";
+import WaitingForQuestion from "../../components/player/WaitingForQuestion";
+import { observer } from "@legendapp/state/react";
+
+// 30 seconds
+const TIMER_DURATION = 30;
+
+const PlayQuiz = observer(() => {
+  // nice hook to simplify the timer logic, default interval: 1000 ms (1 second)
+  const [count, { startCountdown }] = useCountdown({
```

Comment:
[NIT] - consider using legend state for this as it'll reduce re-renders. Not a big deal  all-in-all so feel free to ignore this. From section 3 https://legendapp.com/open-source/state/intro/introduction/